ledger/catchpointwriter.go

```diff
@@ -222,7 +226,7 @@ func (cw *catchpointWriter) WriteStep(ctx context.Context) (more bool, err error
 }
 
 func (cw *catchpointWriter) readDatabaseStep(ctx context.Context, tx *sql.Tx) (err error) {
```

Comment:
Q: must it be called with DB lock? if so, please add a comment

Reply:
are you referring to `accountUpdates.accountsMu` ? if so, then the answer is "no".
The writer is called in exactly two cases:
1. during the startup ( loadFromDisk ), where the writing go-routine has yet to be created. So obviously, in this case there won't be any other resource competing over a lock.
2. during commitRound. Since the commitRound itself is the only database writer, there is no chance that another go-routine would write to the database.

Reply:
> are you referring to accountUpdates.accountsMu

Not sure, was talking about `au.dbs.rdb.Atomic`

Reply:
The `Atomic` afaik, doesn't really take any locks on it's own, unless you're referring to it's variant `AtomicCommitWriteLock`. This is needed only for write locks, since we want to ensure we synchronize the writing to the database to be atomic. However, since the writing is executed on the same goroutine as the catchpoint writer, there is no need to take the lock. ( we can take the read lock, but there are no gains in doing that either )